### PR TITLE
realtime notification through SSE

### PR DESCRIPTION
**Pull request by Maokli**
Adding SSE to notify users realtime of updates within their rental requests

### CODE REVIEW

File: api/src/Rentalcar/rentalcar.service.ts [hunk 1]
```diff
@@ -336,6 +343,22 @@ export class RentalCarService {
...
+          userId: rentalrequest.renterId,
+        });
+      else if(input.newStatus == statusRequest.Cancelled)
+        this.eventEmitter.emit('sse.event', {
+          message: 'Your request was cancelled!',
+          userId: rentalrequest.renterId,
+        });
+        else if(input.newStatus == statusRequest.Paid)
+          this.eventEmitter.emit('sse.event', {
+            message: 'Payment Recieved!',
```
Comment (chamsTmar1) [inline api/src/Rentalcar/rentalcar.service.ts:359]: tiny winy typo on "Received"

File: client/package.json [hunk 2]
```diff
@@ -49,7 +49,7 @@
     "web-vitals": "^2.1.4"
   },
   "scripts": {
-    "start": "set PORT=3000 && react-scripts start",
+    "start": "export PORT=3333 && react-scripts start",
```
Comment (chamsTmar1) [inline client/package.json:52]: Windows 🥲

File: client/src/pages/app/mainApp.wrapper.tsx [hunk 3]
```diff
@@ -11,11 +11,34 @@ import RentalRequestsPage from './rental-requests.page';
 import BrowseUserCars from './userCarsPage';
 import BrowseUserRentalRequests from './rentalrequestPage';
 import UpdateCarForm from '../forms/updateCar.form';
+import { getUserIdFromToken } from '../../services/account.service';
+import { toast } from 'react-toastify';
```
Comment (chamsTmar1) [inline client/src/pages/app/mainApp.wrapper.tsx:15]: Wouldn't it be better if we configured notifications toasters in axios helper as well, then only call axios helper here?

**Review by chamsTmar1 — CHANGES_REQUESTED** [review]
Checkout small comments please (last one especially)

File: client/src/pages/app/mainApp.wrapper.tsx (hunk 3 above)
Comment (Maokli) [reply client/src/pages/app/mainApp.wrapper.tsx:15]: no, the communication mechanism isn't based on HTTP requests so axios cannot be used

File: api/src/Rentalcar/rentalcar.service.ts (hunk 1 above)
Comment (Maokli) [reply api/src/Rentalcar/rentalcar.service.ts]: done

**Review by chamsTmar1 — APPROVED** [review]
GJ

**Review by DogWitDaButter — APPROVED** [review]

**Pull request merged**